api/v1beta1/common_types.go

```diff
@@ -101,6 +101,83 @@ type BarbicanComponentTemplate struct {
 	NetworkAttachments []string `json:"networkAttachments,omitempty"`
 }
 
+// BarbicanLunaTemplate - Variables specific to Thales' Luna HSMs
+type BarbicanLunaTemplate struct {
+        // +kubebuilder:validation:Optional
+        // +kubebuilder:default=4
+        // +kubebuilder:validation:Maximum=7
+        // +kubebuilder:validation:Minimum=0
+        // Level of logging, where 0 means "no logging" and 7 means "debug".
+        LunaLoggingLevel int `json:"lunaLoggingLevel"`
+
+        // +kubebuilder:validation:Optional
+        // The HSM certificates. The map's key is the HSM's IP address and
+        // the value is the OpenShift secret storing the certificate.
+        LunaHSMCertificates map[string]string `json:"lunaHsmCertificates"`
+
+        // +kubebuilder:validation:Optional
+        // The OpenShift secret storing the client certificate plus its key
+        LunaClientCertificate string `json:"lunaClientCertificate"`
+
+        // +kubebuilder:validation:Optional
+        // +kubebuilder:default="/usr/lib/libCryptoki2_64.so"
+	// Path to vendor PKCS11 library
+	LunaLibraryPath string `json:"lunaLibraryPath"`
+
+        // +kubebuilder:validation:Optional
+        // +kubebuilder:default="12345678"
+	// Token serial number used to identify the token to be used.  Required
+	// when the device has multiple tokens with the same label.
+	LunaTokenSerialNumber string `json:"lunaTokenSerialNumber"`
+
+        // +kubebuilder:validation:Optional
+	// Token label used to identify the token to be used.  Required when
+	// token_serial_number is not specified.
+	LunaTokenLabel string `json:"lunaTokenLabel"`
+
+        // +kubebuilder:validation:Optional
+	// The OpenShift secret containing the password to login to PKCS11 session
+	LunaPassword string `json:"lunaPassword"`
```

Comment:
LunaPassword shouldn't be stored here.  We're going to put it in openstack-secret.  Instead you should be adding to the PasswordSelector.

Reply:
In another PR, there was a decision towards storing server and client certificates and key in a separate secret, not in `osp-secret`. What is the final decision then?